python-sdk/src/langwatch/telemetry/context.py

```
@@ -50,10 +62,78 @@
+
+def _reset_current_trace(token: contextvars.Token):
+    global main_thread_langwatch_trace
+    if not _is_on_child_thread():
+        main_thread_langwatch_trace.pop()
+
+    try:
+        stored_langwatch_trace.reset(token)
+    except Exception as e:
+        # Only warn if it's not a context error
+        if "different Context" not in str(e):
+            warnings.warn(f"Failed to reset LangWatch trace context: {e}")
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)